app/services/course/user_registration_service.rb

```diff
@@ -5,15 +5,12 @@ class Course::UserRegistrationService
   # @param [Course::Registration] registration The registration object to be processed.
   # @return [Boolean] True if the registration succeeded. False if the registration failed.
   def register(registration)
-    CourseUser.transaction do
-      course_user = create_or_update_registration(registration)
-      succeeded = course_user && !course_user.changed?
-      if succeeded && course_user.requested?
-        notify_course_staff(registration.course, course_user)
-      else
-        succeeded
-      end
-    end
+    course_user = CourseUser.transaction { create_or_update_registration(registration) }
+
+    succeeded = course_user && !course_user.changed?
+    return succeeded unless course_user.requested?
```

Comment:
I don't know course_user can be nil either > <  I guess that's why it was implemented like that before.